src/channel/ably/auth.ts

```diff
@@ -0,0 +1,139 @@
+import { beforeChannelAttach } from './attach';
+import { httpReqFunction, parseJwt, toTokenDetails } from './utils';
+import { SequentialAuthTokenRequestExecuter } from './token-request';
+import { AblyChannel } from '../ably-channel';
+import { AblyConnector } from '../../connector/ably-connector';
+import { AblyPresenceChannel } from '../ably-presence-channel';
+
+export class AblyAuth {
+
+    // TODO - Can be updated with request throttle, to send multiple request payload under single request
```

Comment:
There's a TODO here, we should address this before we submit the upstream PR

Reply:
We don't really need to do this, this was supposed to be a part of future enhancements. I will remove TODO for now.

Reply:
Updated!